dht.go

```diff
@@ -125,6 +120,16 @@ type IpfsDHT struct {
 
 	autoRefresh bool
 
+	// A function performing a lookup request to a remote peer.ID, verifying that it is able to
+	// answer it correctly
+	lookupCheck         func(context.Context, peer.ID) error
```

Comment:
This can be a method.

Go capture objects when taking function pointers to instanciated methods:
```go
type someType struct{
  v int
}

func (t *someType) doSomething() {
  fmt.Println(t.v)
  t.v++
}

func callDoer(f func()) { f() }

// ...
o := &someType{}
callDoer(o.doSomething) // 0
callDoer(o.doSomething) // 1
callDoer(o.doSomething) // 2
```